Source/Core/DolphinWX/ISOFile.h

```diff
@@ -36,78 +36,101 @@ class GameListItem
   std::string GetName(DiscIO::Language language) const;
   std::string GetName() const;
   std::string GetUniqueIdentifier() const;
   std::string GetDescription(DiscIO::Language language) const;
   std::string GetDescription() const;
   std::vector<DiscIO::Language> GetLanguages() const;
   std::string GetCompany() const { return m_company; }
-  u16 GetRevision() const { return m_Revision; }
+  u16 GetRevision() const { return m_revision; }
   const std::string& GetGameID() const { return m_game_id; }
   u64 GetTitleID() const { return m_title_id; }
   const std::string GetWiiFSPath() const;
   DiscIO::Region GetRegion() const { return m_region; }
-  DiscIO::Country GetCountry() const { return m_Country; }
-  DiscIO::Platform GetPlatform() const { return m_Platform; }
+  DiscIO::Country GetCountry() const { return m_country; }
+  DiscIO::Platform GetPlatform() const { return m_platform; }
   DiscIO::BlobType GetBlobType() const { return m_blob_type; }
-  const std::string& GetIssues() const { return m_issues; }
-  int GetEmuState() const { return m_emu_state; }
-  u64 GetFileSize() const { return m_FileSize; }
-  u64 GetVolumeSize() const { return m_VolumeSize; }
+  const std::string& GetIssues() const { return m_emu_state.issues; }
+  int GetEmuState() const { return m_emu_state.rating; }
+  u64 GetFileSize() const { return m_file_size; }
+  u64 GetVolumeSize() const { return m_volume_size; }
   // 0 is the first disc, 1 is the second disc
   u8 GetDiscNumber() const { return m_disc_number; }
   // NOTE: Banner image is at the original resolution, use WxUtils::ScaleImageToBitmap
   //   to display it
-  const wxImage& GetBannerImage() const { return m_image; }
+  const wxImage& GetBannerImage() const { return m_banner_wx; }
   void DoState(PointerWrap& p);
-  bool ReloadBannerIfNeeded();
+  bool BannerChanged();
+  void BannerCommit();
+  bool EmuStateChanged();
+  void EmuStateCommit();
 
 private:
+  struct EmuState
+  {
+    int rating{};
+    std::string issues{};
+    bool operator!=(const EmuState& rhs) const
+    {
+      return rating != rhs.rating || issues != rhs.issues;
+    }
+    void DoState(PointerWrap& p);
+  };
+  struct Banner
+  {
+    std::vector<u8> buffer{};
+    int width{};
+    int height{};
+    bool empty() const { return buffer.empty(); }
+    void DoState(PointerWrap& p);
+  };
+
   bool IsElfOrDol() const;
-  // Outputs to m_pImage
-  void ReadVolumeBanner(const std::vector<u32>& buffer, int width, int height);
-  // Outputs to m_image
+  void ReadVolumeBanner(std::vector<u8>* image, const std::vector<u32>& buffer, int width,
+                        int height);
+  // Outputs to m_banner_wx
   bool SetWxBannerFromPngFile(const std::string& path);
-  void SetWxBannerFromRaw();
+  void SetWxBannerFromRaw(const Banner& banner);
 
-  // IMPORTANT: All data members must be save/restored in DoState.
+  // IMPORTANT: Nearly all data members must be save/restored in DoState.
```

Comment:
Can we add more information here?  "Nearly all" would not tell me if I need to add a member to be saved or not.  When should this be done?